src/utils/validateNric.tsx

```diff
@@ -35,19 +36,24 @@ const validate = (nricInput: string): boolean => {
   }
 
   // if the nric type is T or G, add 4 to the total
+  // else if the nric type is M, add 3 to the total
   if (["T", "G"].indexOf(nricType) >= 0) {
     total += 4;
+  } else if (["M"].indexOf(nricType) >= 0) {
+    total += 3;
   }
 
   // check last letter of nric for local
   const letterIndex = total % 11;
   const nricLetter = nricArr[3];
   if (["S", "T"].indexOf(nricType) >= 0) {
     return nricLetterST[letterIndex] === nricLetter;
+  } else if (["T", "G"].indexOf(nricType) >= 0) {
```

Comment:
Hm.. feels like typo. Should be `["F", "G"]` instead right? 😬

Reply:
@andrehadianto Please write some tests for this, since the existing tests doesn't catch this mistake 😨

Reply:
replied in the comment above!

Reply:
If it is supposed to be `["T", "G"]`, then when does it ever go to the `else if` branch when it is `T`?

Reply:
got it HAHAH